bin/p2-rm/main.go

```diff
@@ -2,34 +2,65 @@
 package main
 
 import (
-	"bufio"
 	"fmt"
 	"os"
+	"os/user"
 	"path"
-	"strings"
+	"sync"
+	"time"
 
+	"github.com/square/p2/Godeps/_workspace/src/github.com/hashicorp/consul/api"
 	"github.com/square/p2/Godeps/_workspace/src/gopkg.in/alecthomas/kingpin.v2"
 
 	"github.com/square/p2/pkg/kp"
+	"github.com/square/p2/pkg/kp/consulutil"
 	"github.com/square/p2/pkg/kp/flags"
+	"github.com/square/p2/pkg/kp/rcstore"
 	"github.com/square/p2/pkg/labels"
+	"github.com/square/p2/pkg/logging"
 	"github.com/square/p2/pkg/rc"
+	"github.com/square/p2/pkg/rc/fields"
 	"github.com/square/p2/pkg/types"
 	"github.com/square/p2/pkg/version"
 )
 
 // Command line flags
 var (
-	podName  = kingpin.Arg("pod", "The names of the pod to be removed").Required().String()
-	nodeName = kingpin.Flag("node", "The node to do the scheduling on. Uses the hostname by default.").String()
+	podName      = kingpin.Arg("pod", "The names of the pod to be removed").Required().String()
+	nodeName     = kingpin.Flag("node", "The node to do the scheduling on. Uses the hostname by default.").String()
+	deallocation = kingpin.Flag("deallocate", "Specifies that we are deallocating this pod on this node. Using this switch will mutate the desired_replicas value on a managing RC, if one exists.").Bool()
 )
 
+// P2RM contains the state necessary to safely remove a pod from a node
+type P2RM struct {
+	Store       kp.Store
+	ConsulStore rcstore.Store
+	Client      *api.Client
+	Labeler     *labels.ConsulApplicator
```

Comment:
why can't this just be a `labels.Applicator` and then you don't have to export `labels.consulApplicator`